### ISSUE TITLE
MAUI Cross Platform Support

### ISSUE THREAD

**Issue by Todd442** [sunkin351/FmodAudio#11]
First off, very cool library!

Wondering if you've explored generating the necessary abstractions for FMOD to support a cross platform implementation?

I've seen posts that talk about using the native FMOD Libs in that manner.

**Comment by sunkin351** [comment]
This library supports any platform that .NET supports. The only tricky thing I foresee is locating the native binary for FMOD, which I leave up to the OS or the user (via `Fmod.SetLibraryLocation()`)

**Comment by Todd442** [comment]
You're correct but the code as written will need a OS level abstraction for the FMOD Lib. Having used other libs cross platform, I've had the generate these abstractions for the say the Android version of an API contained in .JAR files. When these are created the APIs exposed may not be named the same on the different platforms once the tooling generates the libs required for .NET

So while your library is "technically" cross platform the calls to the FMOD on other platforms require another layer

If that makes any sense

**Comment by sunkin351** [comment]
Quick question, what API's specifically are you talking about that I'm using?

Edit: You're gonna have to explain to me what exactly isn't adequate because I'm completely unfamiliar with android development.

**Comment by sunkin351** [comment]
After reviewing your comment further, yes, if the naming scheme of the native methods changes, everything breaks. Idk what I can really do about that.

**Comment by Todd442** [comment]
The core here is you'd need to include the type abstraction as part of your NUGET and deal with it via compile directives.

**Comment by sunkin351** [comment]
Maybe I'm stupid, but I'm not following... Could you clarify that further?

**Comment by sunkin351** [comment]
Closed until a solid plan of action can be provided.

**Comment by Todd442** [comment]
Would like to reopen this after more research.

First a couple clarifications. 

**In my earlier comment I said the namespaces could be affected by the generation of a "Android Java Library Binding" I was incorrect. The namespace and interfaces are unaffected.** 

There's a simple walk though for this for another FMOD library (different functionality) found at the link below so you can see the process.

https://github.com/Martenfur/FmodForFoxes

I have followed a similar process with your library and am now receiving an "An error occurred that wasn't supposed to.  Contact support."
  
![image](https://user-images.githubusercontent.com/7561232/232528750-3b308b60-717a-4138-8285-ad6cff4c5482.png)

![image](https://user-images.githubusercontent.com/7561232/232529167-90e2fe1a-0d6e-42fa-a261-a8481c894451.png)

![image](https://user-images.githubusercontent.com/7561232/232529599-820e2def-408a-4b04-b9b6-8ab87d4a5b49.png)

I can share the code if you'd like to try and resolve the issue.  

I have confirmed that this is not related to the libraries not being found.

**Comment by sunkin351** [comment]
Could you confirm what Error code is being returned from `System_Create()`?

Also, sorry for the late reply, I've been busy of late.
